api/python/t4/util.py

```diff
@@ -366,3 +366,19 @@ def quiltignore_filter(paths, ignore, url_scheme):
         return files.union(dirs)
     else:
         raise NotImplementedError
+
+def validate_key(key):
+    """
+    Verify that a file path or S3 path does not contain any '.' or '..' separators or files.
+    """
+    if key is None or key == '':
+        raise QuiltException(
+            f"Invalid key {key!r}. A package entry key cannot be empty."
+        )
+
+    if ((key == '.' or '/./' in key or key.startswith('./') or key.endswith('/.')) or
```

Comment:
That's kind of ugly... Do something like this instead: https://github.com/quiltdata/t4/blob/master/api/python/t4/data_transfer.py#L513

Or even just move that function into `util.py`

Reply:
Done.